#VRML_SIM R2020b utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.5394617034192034 -0.6721106453422985 -0.5071965604788842 1.98079194139074
  position -2.6444261915308713 8.407473690734474 -0.017066745215178675
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 5 6
  wallThickness 0.1
  wallHeight 0.5
}
DEF robot0 Robot {
  translation -2 0.0397206 1.63779
  rotation 5.215639999621358e-06 0.9999999999274027 1.0862399999211419e-05 1.5708
  children [
    Emitter {
      channel -1
    }
    Receiver {
      channel -1
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.9661830314
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 1.9663486467011823e-17 0.049999999999999996
        rotation -0.6402252937858931 0.6397156681170345 0.42529452989939 2.3367787169696017
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.029349967685
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 -2.0857802546691148e-17 0.05
        rotation 0.13858446151637724 -0.1384741469137228 0.9806218729269364 1.5895672418624276
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.96646596042
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 -1.9663115813363407e-17 -0.049999999999999996
        rotation -0.6401897561460392 0.639680158765494 0.42540142290244787 2.3365975836836315
        children [
          USE Wheel
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02969236843
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 -1.215722899994343e-08 -0.05000000350782249
        rotation 0.13874912168836348 -0.13863867601458071 0.9805753406769068 1.589614686384032
        children [
          USE Wheel
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    DistanceSensor {
      translation 0.02 0 0.1
      rotation 0 1 0 -1.27
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0 0.1
      rotation 0 1 0 -1.87
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
    GPS {
      translation 0 0.03 0
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.02
                radius 0.01
              }
            }
          ]
        }
      ]
      name "global"
    }
    InertialUnit {
      translation 0 0.03 0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.7
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "imu"
    }
  ]
  name "robot0_name"
  boundingObject USE Body
  physics Physics {
  }
  controller "robot_controller_v2"
  controllerArgs [
    "0"
  ]
  supervisor TRUE
}
DEF robot1 Robot {
  translation -2 0.0397206 0.617791
  rotation 5.215639999621358e-06 0.9999999999274027 1.0862399999211419e-05 1.5708
  children [
    Emitter {
      channel -1
    }
    Receiver {
      channel -1
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.96618303254
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 1.966348646551623e-17 0.049999999999999996
        rotation -0.6402252936424335 0.6397156679736891 0.42529453033096537 2.336778716238256
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02934996494
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 -2.085780244765497e-17 0.05
        rotation 0.13858446019621606 -0.13847414559461266 0.9806218732997777 1.5895672414822846
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.96646595928
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 -1.9663115814841905e-17 -0.049999999999999996
        rotation -0.6401897562886323 0.639680158907973 0.42540142247361173 2.3365975844102778
        children [
          USE Wheel
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02969237117
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 -1.215722899036132e-08 -0.05000000350782252
        rotation 0.13874912300840056 -0.13863867733356686 0.9805753403036397 1.589614686764625
        children [
          USE Wheel
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    DistanceSensor {
      translation 0.02 0 0.1
      rotation 0 1 0 -1.27
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0 0.1
      rotation 0 1 0 -1.87
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
    GPS {
      translation 0 0.03 0
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.02
                radius 0.01
              }
            }
          ]
        }
      ]
      name "global"
    }
    InertialUnit {
      translation 0 0.03 0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.7
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "imu"
    }
  ]
  name "robot1_name"
  boundingObject USE Body
  physics Physics {
  }
  controller "robot_controller_v2"
  controllerArgs [
    "1"
  ]
  supervisor TRUE
}
DEF robot2 Robot {
  translation -2 0.0397206 -0.632209
  rotation 5.215639999621358e-06 0.9999999999274027 1.0862399999211419e-05 1.5708
  children [
    Emitter {
      channel -1
    }
    Receiver {
      channel -1
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.96618303254
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 1.6751396224103286e-17 0.04999999999999999
        rotation -0.6402252936424339 0.6397156679736886 0.4252945303309654 2.336778716238256
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02934996494
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 1.3204848660514083e-18 0.05000000000000001
        rotation 0.13858446019621679 -0.13847414559461335 0.9806218732997776 1.589567241482283
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.96646595928
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 -1.6753334272846483e-17 -0.04999999999999999
        rotation -0.6401897562886323 0.6396801589079728 0.42540142247361157 2.336597584410277
        children [
          USE Wheel
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02969237117
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 -1.215722899562054e-08 -0.05000000350782252
        rotation 0.13874912300840092 -0.13863867733356713 0.9805753403036397 1.5896146867646241
        children [
          USE Wheel
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    DistanceSensor {
      translation 0.02 0 0.1
      rotation 0 1 0 -1.27
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0 0.1
      rotation 0 1 0 -1.87
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
    GPS {
      translation 0 0.03 0
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.02
                radius 0.01
              }
            }
          ]
        }
      ]
      name "global"
    }
    InertialUnit {
      translation 0 0.03 0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.7
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "imu"
    }
  ]
  name "robot2_name"
  boundingObject USE Body
  physics Physics {
  }
  controller "robot_controller_v2"
  controllerArgs [
    "2"
  ]
  supervisor TRUE
}
DEF robot3 Robot {
  translation -2 0.0397206 -1.61221
  rotation 5.215639999621358e-06 0.9999999999274027 1.0862399999211419e-05 1.5708
  children [
    Emitter {
      channel -1
    }
    Receiver {
      channel -1
    }
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 0 1 0
        metalness 0
      }
      geometry Box {
        size 0.1 0.05 0.2
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.9661830314
        anchor 0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 1.966348646699534e-17 0.049999999999999996
        rotation -0.6402252937858931 0.6397156681170345 0.42529452989939 2.3367787169696017
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 1 0 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02934996769
        anchor -0.06 0 0.05
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation -0.06 -2.0938429362928983e-17 0.05
        rotation 0.138584461519879 -0.13847414691722168 0.9806218729259474 1.5895672418634357
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 43816.96646596042
        anchor 0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation 0.060000000000000026 -1.966311581334721e-17 -0.049999999999999996
        rotation -0.6401897561460393 0.6396801587654937 0.42540142290244787 2.3365975836836315
        children [
          USE Wheel
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 40376.02969236842
        anchor -0.06 0 -0.05
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.06 -1.2157229000162206e-08 -0.05000000350782249
        rotation 0.1387491216848623 -0.13863867601108235 0.9805753406778968 1.5896146863830223
        children [
          USE Wheel
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    DistanceSensor {
      translation 0.02 0 0.1
      rotation 0 1 0 -1.27
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
    }
    DistanceSensor {
      translation -0.02 0 0.1
      rotation 0 1 0 -1.87
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
    }
    GPS {
      translation 0 0.03 0
      children [
        Solid {
          children [
            Shape {
              appearance PBRAppearance {
              }
              geometry Cylinder {
                height 0.02
                radius 0.01
              }
            }
          ]
        }
      ]
      name "global"
    }
    InertialUnit {
      translation 0 0.03 0.04
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0.7
            metalness 0
          }
          geometry Box {
            size 0.02 0.02 0.02
          }
        }
      ]
      name "imu"
    }
  ]
  name "robot3_name"
  boundingObject USE Body
  physics Physics {
  }
  controller "robot_controller_v2"
  controllerArgs [
    "3"
  ]
  supervisor TRUE
}
DEF mine0 Robot {
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "mine0_name"
}
DEF mine1 Robot {
  translation 1.61 0 0.91
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "mine1_name"
}
DEF mine2 Robot {
  translation -1.24 0 0.63
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "mine2_name"
}
DEF mine3 Robot {
  translation -1.4 0 -1.35
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "mine3_name"
}
DEF mine4 Robot {
  translation 0.88 0 -0.35
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "mine4_name"
}
DEF mine5 Robot {
  translation -0.37 0 1.36
  children [
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
      }
      geometry Cylinder {
        height 0.5
        radius 0.05
      }
    }
  ]
  name "mine5_name"
}
